code/datums/components/deployable.dm

```diff
@@ -0,0 +1,63 @@
+/datum/component/deployable
+	var/deploy_time = 5 SECONDS //Default time it takes to deploy the item
+	var/obj/thing_to_be_deployed //What do we spawn when the item is deployed successfully?
+	var/deployed_name //For getting the name of an object for examines later on
+	var/delete_on_use = TRUE //Do we delete the item being used when the object is deployed
+
+/datum/component/deployable/Initialize(thing_to_be_deployed)
+	. = ..()
+	if(!isitem(parent))
+		return COMPONENT_INCOMPATIBLE
+
+	src.thing_to_be_deployed = thing_to_be_deployed
+	src.delete_on_use = delete_on_use
+
+	RegisterSignal(parent, COMSIG_PARENT_EXAMINE, .proc/examine)
+	RegisterSignal(parent, COMSIG_ITEM_ATTACK_SELF, .proc/deploy)
+
+	var/obj/item/typecast = thing_to_be_deployed
+	deployed_name = initial(typecast.name)
+
+/datum/component/deployable/proc/examine(datum/src, mob/user, list/examine_list)
+	SIGNAL_HANDLER
+
+	if(thing_to_be_deployed) //Tells the player what this item can deploy into
```

Comment:
don't bother checking this, if thing_to_be_deployed is false the component should be throwing errors as soon as its added.

Reply:
this is here simply because admins can vv the component, and its always possible for admins to mess up setting thing_to_be_deployed

Reply:
coding with the mindset of "admin can var-edit" is horrendous for items like this and makes the code quality way worse because we'd need sanity checks for everything in every proc.